pandas/core/index.py

```diff
@@ -3878,6 +3878,32 @@ def _convert_slice_indexer(self, key, kind=None):
         # translate to locations
         return self.slice_indexer(key.start, key.stop, key.step)
 
+    def _format_native_types(self, na_rep='', float_format=None,
+                             decimal='.', quoting=None, **kwargs):
```

Comment:
since this is basically identical to ``core/internals/FloatBlock/to_native_types``.  let's pull both those out and put it in a function in ``core/format.py/FloatArrayFormatter`` and call it ``.get_formatted_data()``. See how that works out. These are the routines for screen printing (which are necessarily different from to_csv / index formatting).